point

```diff
@@ -197,7 +205,74 @@ if (program.migrate) {
     return;
 }
 
+// ------------------ Remove Everything ------------ //
+
+if (program.debug_destroy_everything) {
+    // todo: remove in prod, dangerous
+    ctx.db = new DB(ctx);
+    DB.__debugClearCompletely(ctx); // async!
+    return;
+}
+
+// ------------------ Compile Contracts ------------ //
+
+const CONTRACT_ADDRESSES = ['Identity', 'Migrations', 'StorageProviderRegistry'];
+
+if (program.compile) {
+    const getImports = dependency => {
+        const dependencyNodeModulesPath = path.join(__dirname, 'node_modules/', dependency);
+        if (!existsSync(dependencyNodeModulesPath)) {
+            throw new Error(
+                `Could not find contract dependency "${dependency}", have you tried npm install?`
+            );
+        }
+        return {contents: readFileSync(dependencyNodeModulesPath, 'utf8')};
+    };
+
+    log.info('Compiling contracts...');
+
+    const buildPath = path.resolve(__dirname, 'truffle', 'build', 'contracts');
+    if (!existsSync(buildPath)) {
+        mkdirSync(buildPath);
+    }
+    const contractsPath = path.resolve(__dirname, 'truffle', 'contracts');
+
+    Promise.all(CONTRACT_ADDRESSES.map(async name => {
```

Comment:
This solution is based on a slightly outdated version of the `compileContracts` function. Please see the most recent one [here](https://github.com/pointnetwork/pointnetwork/blob/develop/scripts/patch-config.js#L25). The main difference is that we don't want to re-compile the contracts if they are already compiled. The corresponding test takes place [here](https://github.com/pointnetwork/pointnetwork/blob/develop/scripts/patch-config.js#L41). Please double check for any other updates to the function code.